mec/mec1501/component/spi_periph.h

```diff
@@ -0,0 +1,341 @@
+#define MCHP_SPIP_EC_IEN_RXF_SZ		(1ul << 24)
+#define MCHP_SPIP_EC_IEN_TXF_UNF	(1ul << 25)
+#define MCHP_SPIP_EC_IEN_TXF_OVF	(1ul << 26)
+#define MCHP_SPIP_EC_IEN_RXF_UNF	(1ul << 27)
+#define MCHP_SPIP_EC_IEN_RXF_OVF	(1ul << 28)
+
+/* SPIP Memory Config register */
+#define MCHP_SPIP_MCFG_REG_OFS		0x14
+#define MCHP_SPIP_MCFG_REG_MASK		0x03ul
+#define MCHP_SPIP_MCFG_BAR0_EN_POS	0
+#define MCHP_SPIP_MCFG_BAR0_EN		(1ul << 0)
+#define MCHP_SPIP_MCFG_BAR1_EN_POS	1
+#define MCHP_SPIP_MCFG_BAR1_EN		(1ul << 1)
+
+/* SPIP Memory Base Address 0 register */
+#define MCHP_SPIP_MBA0_REG_OFS		0x18
+#define MCHP_SPIP_MBA0_REG_MASK		0xFFFFFFFCul
+
+/* SPIP Memory Write Limit 0 register */
+#define MCHP_SPIP_MWLIM0_REG_OFS	0x1C
+#define MCHP_SPIP_MWLIM0_REG_MASK	0xFFFFFFFCul
+
+/* SPIP Memory Read Limit 0 register */
+#define MCHP_SPIP_MRLIM0_REG_OFS	0x20
+#define MCHP_SPIP_MRLIM0_REG_MASK	0xFFFFFFFCul
+
+/* SPIP Memory Base Address 1 register */
+#define MCHP_SPIP_MBA1_REG_OFS		0x24
+#define MCHP_SPIP_MBA1_REG_MASK		0xFFFFFFFCul
+
+/* SPIP Memory Write Limit 1 register */
+#define MCHP_SPIP_MWLIM1_REG_OFS	0x28
+#define MCHP_SPIP_MWLIM1_REG_MASK	0xFFFFFFFCul
+
+/* SPIP Memory Read Limit 1 register */
+#define MCHP_SPIP_MRLIM1_REG_OFS	0x2C
+#define MCHP_SPIP_MRLIM1_REG_MASK	0xFFFFFFFCul
+
+/* SPIP RX FIFO Host BAR register */
+#define MCHP_SPIP_RXFHB_REG_OFS	0x30
+#define MCHP_SPIP_RXFHB_REG_MASK	0xFFFFul
+
+/* SPIP RX FIFO Host BAR register */
+#define MCHP_SPIP_RXFBC_REG_OFS	0x34
+#define MCHP_SPIP_RXFBC_REG_MASK	0x7FFFul
+
+/* SPIP TX FIFO Host BAR register */
+#define MCHP_SPIP_TXFHB_REG_OFS	0x38
+#define MCHP_SPIP_TXFHB_REG_MASK	0xFFFFul
+
+/* SPIP TX FIFO Host BAR register */
+#define MCHP_SPIP_TXFBC_REG_OFS	0x3C
+#define MCHP_SPIP_TXFBC_REG_MASK	0x7FFFul
+
+/* SPIP System Configuration register */
+#define MCHP_SPIP_SYSCFG_REG_OFS	0x40
+#define MCHP_SPIP_SYSCFG_REG_MASK	0x000F04FFul
+#define MCHP_SPIP_SYSCFG_SRST		(1ul << 0)
+#define MCHP_SPIP_SYSCFG_LOCK_WM	(1ul << 1)
+#define MCHP_SPIP_SYSCFG_LOCK_TAR	(1ul << 2)
+#define MCHP_SPIP_SYSCFG_LOCK_WC	(1ul << 3)
+#define MCHP_SPIP_SYSCFG_LOCK_SPI_STS	(1ul << 4)
+#define MCHP_SPIP_SYSCFG_LOCK_SPI_IEN	(1ul << 5)
+#define MCHP_SPIP_SYSCFG_LOCK_MBA0	(1ul << 6)
+#define MCHP_SPIP_SYSCFG_LOCK_MBA1	(1ul << 7)
+#define MCHP_SPIP_SYSCFG_LOCK_TST	(1ul << 10)
+#define MCHP_SPIP_SYSCFG_LOCK_ALL	0x04FEul
+#define MCHP_SPIP_SYSCFG_ACT		(1ul << 16)
+#define MCHP_SPIP_SYSCFG_MASK_EC	(1ul << 17)
+#define MCHP_SPIP_SYSCFG_SIMPLE_MODE	(1ul << 18)
+#define MCHP_SPIP_SYSCFG_ECDA		(1ul << 19)
+
+/* SPIP SPI Master to EC Mailbox register */
+#define MCHP_SPIP_MB_S2EC_REG_OFS	0x44
+#define MCHP_SPIP_MB_S2EC_REG_MASK	0xFFFFFFFFul
+#define MCHP_SPIP_MB_S2EC_CLR		0xFFFFFFFFul
+
+/* SPIP EC to SPI Master Mailbox register */
+#define MCHP_SPIP_MB_EC2S_REG_OFS	0x48
+#define MCHP_SPIP_MB_EC2S_REG_MASK	0xFFFFFFFFul
+#define MCHP_SPIP_MB_EC2S_CLR		0xFFFFFFFFul
+
+/* SPI commands supported by SPIP */
+#define SPIP_CMD_IN_BAND_RST		0xFFu
```

Comment:
Please drop these 'u' things...they are just confusing

Reply:
@nashif are we now dropping the misra standard rule rule 7.2 from Microchip HAL too?

Reply:
Outside of the Zephyr tree this is up to the owner of the code. We only enforce guidelines on Zephyr code, so it is up to you/michrochip. Please change this to be a 'U' instead and use capitals everywhere to be consistent.

Reply:
7.2 is still not enforced in Zephyr but it is part of the guidelines, if changes are requested on something like this, just reference the coding guidelines.

Reply:
@nashif 
I think the **optional** aspect for vendor modules/HALs following misra should be captured in code guidelines. There is nothing there I find this on case-by-case basis confusing.  https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#coding-guidelines 

@sjg20 @scottwcpg it would be very confusing to have this update of MEC15xx HAL without the misra guidelines while the rest of MEC15xx is already following it.

Reply:
> I think the **optional** aspect for vendor modules/HALs following misra should be captured in code guidelines.

this was implied because non of the vendor HALs was ever written specifically for Zephyr, so we have no business enforcing guidelines on 3rd party code. If this code here is all zephyr specific, then it is probably was a good idea to start moving this to the zephyr tree and have everything live in one place.
Will look into making sure this is documented to avoid confusion.

Reply:
> @sjg20 @scottwcpg it would be very confusing to have this update of MEC15xx HAL without the misra guidelines while the rest of MEC15xx is already following it.

The code above is not wrong, if we are debating style preferences I think you (@scottwcpg) should make the call how to deal with this request. I personally think it should be left as is with the minor change of lower vs upper case.

Reply:
I will change the files to be more consistent.  "U" for unsigned constants.